lib/migrate.js

```diff
@@ -0,0 +1,444 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: loopback-connector-couchdb2
+// This file is licensed under the Apache License 2.0.
+// License text available at https://opensource.org/licenses/Apache-2.0
+
+'use strict';
+
+var async = require('async');
+var assert = require('assert');
+var g = require('strong-globalize')();
+var util = require('util');
+var _ = require('lodash');
+
+module.exports = mixinMigrate;
+
+function mixinMigrate(CouchDB) {
+  var debug = require('debug')('loopback:connector:couchdb:migrate');
+
+  /**
+ * Perform automigrate for the given models.
+ * - destroys model data if the model exists
+ * - delete existing indexes in database
+ * - create new indexes according to model definition
+ * 
+ * @param {String|String[]} [models] A model name or an array of model names.
+ * If not present, apply to all models
+ * @callback {Function} cb The callback function
+ */
+  CouchDB.prototype.automigrate = function(models, cb) {
+    debug('CouchDB.prototype.automigrate models %j', models);
+    var self = this;
+    var existingModels = models;
+    async.series([
+      function(callback) {
+        destroyData(callback);
+      },
+      function(callback) {
+        debug('CouchDB.prototype.automigrate update indexes for %j', models);
+        self.migrateOrUpdateIndex(models, true, callback);
+      }], cb);
+    function destroyData(destroyCb) {
+      async.eachSeries(existingModels, function(model, cb) {
+        self.destroyAll(model, {}, {}, cb);
+      }, function(err) {
+        debug('CouchDB.prototype.automigrate destroy all data: %j', err);
```

Comment:
I added those debug code when fixing a bug, while seems they are not that well organized/phrased... I may create a quick PR to fix them.